File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        answer = ""
        if "" in strs:
            return ""
        if len(strs) == 1:
            return strs[0]

        i = 0
        prefix = strs[0][i]
        while 1:
            for j in range(len(strs)):
                if i == len(strs[j]):
                    return answer
                if prefix != strs[j][i]:
                    return answer

            answer += strs[0][i]
            
            if i < len(strs[0])-1:
                i += 1
                prefix = strs[0][i]
            else:
                break
        
        return answer
        

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int minLen = 999999;
        for (int i = 0; i < strs.length; i++){
            minLen = (strs[i].length() <  minLen) ? strs[i].length() : minLen;
        }
        for (int i = 0; i < minLen; i++){
            for (int j = 1; j < strs.length; j++){
                if (strs[j].charAt(i) != strs[j-1].charAt(i)){
                    return strs[0].substring(0, i);
                }
            }
        }
        return strs[0].substring(0, minLen);
    }
}